### PR TITLE
Initial ConnectionBuilder path for Oracle Kerberos

### LINKED ISSUE #18121: Add support for Connection Builder path for Oracle Kerberos connections
In some cases, there is an issue between the Oracle driver and IBM Java SDK when running under the DataSource.getConnection() path. This can be mitigated by running under the DataSource.getConnectionBuilder() path.

### PR DESCRIPTION
**Pull request by mswatosh**
Adds new property to allow for using the OracleConnectionBuilder to create Connections with Kerberos.
fixes #18121

### CODE REVIEW

File: dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/OracleHelper.java [hunk 1]
```diff
@@ -857,6 +860,21 @@ public Connection getConnectionFromDatasource(DataSource ds, KerbUsage useKerb,
...
+            
+            //Workaround for Oracle Driver + IBM JDK issue with non-connection builder path
+            if (mcf.dsConfig.get().sendGSSCredentialOnOracleBuilder) {
+                try {
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
+                        Tr.debug(this, tc, "Using Connection Builder path for Kerberos");
+                    Object oracleConnectionBuilder = ds.getClass().getMethod("createConnectionBuilder").invoke(ds);
+                    Object oracleConnectionBuilder2 = oracleConnectionBuilder.getClass().getMethod("gssCredential", GSSCredential.class).invoke(oracleConnectionBuilder, gssCredential);
+                    Method m = oracleConnectionBuilder.getClass().getMethod("build");
+                    m.setAccessible(true);
```
Comment (njr-11) [inline dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/OracleHelper.java]: Why would this need setAccessible(true)?  Whether it is Oracle vendor-specific API or JDBC 4.3 spec API, the method we are invoking ought to be public and should not require this.

File: dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/OracleHelper.java [hunk 2]
```diff
@@ -874,6 +892,27 @@ public ConnectionResults getPooledConnection(final CommonDataSource ds, String u
...
+                        Tr.debug(this, tc, "Using Connection Builder path for Kerberos");
+                    Object oracleConnectionBuilder;
+                    if (is2Phase) { //XAConnection
+                        oracleConnectionBuilder = ds.getClass().getMethod("createXAConnectionBuilder").invoke(ds);
+                    } else { //PooledConnection
+                        oracleConnectionBuilder = ds.getClass().getMethod("createPooledConnectionBuilder").invoke(ds);    
+                    }
+                    Object oracleConnectionBuilder2 = oracleConnectionBuilder.getClass().getMethod("gssCredential", GSSCredential.class).invoke(oracleConnectionBuilder, gssCredential);
+                    Method m = oracleConnectionBuilder.getClass().getMethod("build");
+                    m.setAccessible(true);
```
Comment (njr-11) [inline dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/OracleHelper.java]: same comment as above with setAccessible

**Review by njr-11 — APPROVED** [review]
Some minor comments about unnecessary code, otherwise looks good.

**Comment by mswatosh** [comment]
Added the doPriv, and created a real test case.

**Comment by mswatosh** [comment]
#build

**Review by njr-11 — APPROVED** [review]

**Comment by mswatosh** [comment]
#build

**Comment by mswatosh** [comment]
#libbybot

**Pull request merged**